spec/lib/cyclonedx/report_rust_crates_cyclonedx_spec.rb

```diff
@@ -0,0 +1,213 @@
+require_relative '../../spec_helper'
+require 'json'
+
+describe Cyclonedx::ReportRustCrates do
+  describe "#run" do
+    it 'should report all the deps in the Cargo.lock if manifest is absent in cyclonedx' do
+      repo = Salus::Repo.new('spec/fixtures/report_rust_crates/lock_only')
+      scanner = Salus::Scanners::ReportRustCrates.new(repository: repo, config: {})
+      scanner.run
+
+      rust_cyclonedx = Cyclonedx::ReportRustCrates.new(scanner.report)
+      expect(rust_cyclonedx.build_components_object[0..2]).to match_array(
+        [
+          {
+            "bom-ref": "pkg:cargo_lock/autocfg@0.1.7",
+            "type": "library",
+            "group": "",
+            "name": "autocfg",
+            "version": "0.1.7",
+            "purl": "pkg:cargo_lock/autocfg@0.1.7",
+            "properties": [
+              {
+                "key": "source",
+                "value": "registry+https://github.com/rust-lang/crates.io-index"
+              },
+              {
+                "key": "dependency_file",
+                "value": "Cargo.lock"
+              }
+            ]
+          },
+          {
+            "bom-ref": "pkg:cargo_lock/bitflags@1.2.1",
+            "type": "library",
+            "group": "",
+            "name": "bitflags",
+            "version": "1.2.1",
+            "purl": "pkg:cargo_lock/bitflags@1.2.1",
+            "properties": [
+              {
+                "key": "source",
+                "value": "registry+https://github.com/rust-lang/crates.io-index"
+              },
+              {
+                "key": "dependency_file",
+                "value": "Cargo.lock"
+              }
+            ]
+          },
+          {
+            "bom-ref": "pkg:cargo_lock/cloudabi@0.0.3",
+            "type": "library",
+            "group": "",
+            "name": "cloudabi",
+            "version": "0.0.3",
+            "purl": "pkg:cargo_lock/cloudabi@0.0.3",
+            "properties": [
+              {
+                "key": "source",
+                "value": "registry+https://github.com/rust-lang/crates.io-index"
+              },
+              {
+                "key": "dependency_file",
+                "value": "Cargo.lock"
+              }
+            ]
+          }
+        ]
+      )
+    end
+
+    it 'should calculate and report all the dependencies in the manifest if Cargo.lock is absent' do
+      repo = Salus::Repo.new('spec/fixtures/report_rust_crates/manifest_only')
+      scanner = Salus::Scanners::ReportRustCrates.new(repository: repo, config: {})
+
+      # We will stub the lock file generate to keep our specs from needing to
+      # hit the internet and pulldown the dependency graph
```

Comment:
Thanks for the comment 💯 Maybe add to it that cargo tree will create a lock file if not already present for additional clarity